.travis.yml

```diff
@@ -1,11 +1,27 @@
-sudo: false
+sudo: required
 language: python
-python: 3.5
 
-before_install: pip install -U 'pip>=8'
-script: py.test
+install: docker build -t normandy:build .
+script: docker run --net host normandy:build py.test
+after_success:
+  - '[ -z "$DOCKER_EMAIL" ] ||
+    docker login -e="$DOCKER_EMAIL" -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"'
+  - '[ -z "$DOCKER_EMAIL" ] ||
+    ([ "$TRAVIS_BRANCH" == master ] && TAG=latest || TAG="$TRAVIS_BRANCH" &&
+    docker tag normandy:build "mozilla/normandy:$TAG" &&
+    docker push "mozilla/normandy:$TAG")'
+  - '[ -z "$DOCKER_EMAIL" -o -z "$TRAVIS_TAG" ] ||
+    (docker tag normandy:build "mozilla/normandy:$TRAVIS_TAG" &&
+    docker push "mozilla/normandy:$TRAVIS_TAG")'
```

Comment:
How would you feel about putting all of this `after_success` hook into something like `bin/deploy/dockerhub.sh`?